dbod/api/instance.py

```diff
@@ -109,20 +128,51 @@ def put(self, name):
             if response.ok:
                 response = requests.post(config.get('postgrest', 'volume_url'), json=volumes)
                 if response.ok:
+                    logging.debug("Inserting volumes: " + json.dumps(volumes))
                     self.set_status(response.status_code)
                 else:
-                    logging.error("Error adding volumes for instance: " + str(entid))
+                    logging.error("Error adding volumes for instance: " + response.text)
                     raise tornado.web.HTTPError(response.status_code)
             else:
-                logging.error("Error deleting old volumes for instance: " + str(entid))
+                logging.error("Error deleting old volumes for instance: " + response.text)
                 raise tornado.web.HTTPError(response.status_code)
+                
+        # Check if the attributes are changed
+        if "attributes" in instance:
+            attributes = instance["attributes"]
+            response = requests.delete(config.get('postgrest', 'attribute_url') + "?instance_id=eq." + str(entid))
+            if response.ok:
```

Comment:
This logic is wrong. If there are no attributes present (as during instance creation) `response.ok` will be False, and an exception will be raised when actually that's exactly the opposite behavior from what is expected during and instance creation (when the instance has been partially registered from the webapp)